src/Rings/orderings.jl

```diff
@@ -876,6 +877,70 @@ function is_mixed(M::MonomialOrdering)
   return !isempty(globals) && !isempty(locals)
 end
 
+
+# ex: nvars = 7, sigmaC = {    2, 3,       6   }
+#             =>  sigma = { 1,       4, 5,    7}
+#                varmap = {-1,+1,+2,-2,-3,+3,-4}
+function _elimination_data(n::Int, sigmaC::Vector)
+  varmap = zeros(Int, n)
+  for si in sigmaC
+    varmap[si] = 1
+  end
+  sigma = Int[]
+  sigmaC = Int[]
+  for i in 1:n
+    if varmap[i] == 0
+      push!(sigma, i)
+      varmap[i] = -length(sigma)
+    else
+      push!(sigmaC, i)
+      varmap[i] = +length(sigmaC)
+    end
+  end
+  return varmap, sigma, sigmaC
+end
+
+@doc Markdown.doc"""
+    is_elimination_ordering(M::MonomialOrdering, v::Vector{Int})
+    is_elimination_ordering(M::MonomialOrdering, v::Vector{<:MPolyElem})
+
+Return `true` if the given ordering eliminates the variables in `v`.
+"""
+function is_elimination_ordering(o::MonomialOrdering, sigmaC::Vector{Int})
+# This is probably correct: o is elimination ordering for sigma^C <=>
+#   For each j in sigma^C, the first non-zero weight M[i,j] in column j is >= 0,
+#   and the weights in columns corresponding to sigma in rows <= i are zero.
```

Comment:
@hannes14 Is this a correct characterization of elimination orderings by their weight matrices? Does the singular kernel have such a function somewhere?

Reply:
one needs:
- M is of rank n (number of variables), i.e. "o" is an ordering
- first non-zero weight M[i,j] in each column j is >0, i.e. "o" is a well ordering
- elimination for a subset of variables: in the first row of M the entries corresponding to the to eliminated variables
  must be >0, all the other entries must be zero
Singular does not have a function to test "is_elimination_ordering", but can (via prepending a(v)) change every well ordering to an elimination ordering for these variables where a one is in v.